source/core/Ky.ts

```diff
@@ -105,43 +105,54 @@ export class Ky {
 				throw new RangeError(`The \`timeout\` option cannot be greater than ${maxSafeTimeout}`);
 			}
 
-			await delay(1);
-			let response = await ky._fetch();
-
-			for (const hook of ky._options.hooks.afterResponse) {
-				// eslint-disable-next-line no-await-in-loop
-				const modifiedResponse = await hook(
-					ky.request,
-					ky._options as NormalizedOptions,
-					ky._decorateResponse(response.clone())
-				);
-
-				if (modifiedResponse instanceof globalThis.Response) {
-					response = modifiedResponse;
-				}
-			}
+			return new Promise<Response>((resolve, reject) => {
+				// Fix #345: We must keep setTimeout() here instead of using delay(), because new relic cannot handle parallel promise correctly.
+				setTimeout(async () => {
+					try {
+						let response = await ky._fetch();
+						for (const hook of ky._options.hooks.afterResponse) {
+						// eslint-disable-next-line no-await-in-loop
+							const modifiedResponse = await hook(
+								ky.request,
+								ky._options as NormalizedOptions,
+								ky._decorateResponse(response.clone())
+							);
+
+							if (modifiedResponse instanceof globalThis.Response) {
+								response = modifiedResponse;
+							}
+						}
 
-			ky._decorateResponse(response);
+						ky._decorateResponse(response);
 
-			if (!response.ok && ky._options.throwHttpErrors) {
-				throw new HTTPError(response, ky.request, (ky._options as unknown) as NormalizedOptions);
-			}
+						if (!response.ok && ky._options.throwHttpErrors) {
+							reject(new HTTPError(response, ky.request, (ky._options as unknown) as NormalizedOptions));
+							return;
+						}
 
-			// If `onDownloadProgress` is passed, it uses the stream API internally
-			/* istanbul ignore next */
-			if (ky._options.onDownloadProgress) {
-				if (typeof ky._options.onDownloadProgress !== 'function') {
-					throw new TypeError('The `onDownloadProgress` option must be a function');
-				}
+						// If `onDownloadProgress` is passed, it uses the stream API internally
+						/* istanbul ignore next */
+						if (ky._options.onDownloadProgress) {
+							if (typeof ky._options.onDownloadProgress !== 'function') {
+								reject(new TypeError('The `onDownloadProgress` option must be a function'));
+								return;
+							}
 
-				if (!supportsStreams) {
-					throw new Error('Streams are not supported in your environment. `ReadableStream` is missing.');
-				}
+							if (!supportsStreams) {
+								reject(new Error('Streams are not supported in your environment. `ReadableStream` is missing.'));
+								return;
+							}
 
-				return ky._stream(response.clone(), ky._options.onDownloadProgress);
-			}
+							resolve(ky._stream(response.clone(), ky._options.onDownloadProgress));
+							return;
+						}
 
-			return response;
+						resolve(response);
+					} catch (error: unknown) {
+						reject(error);
+					}
+				}, 1);
```

Comment:
Didn't change 1 to 0, though 0 looks also work

Reply:
I don't remember why it's `1` and not `0`.

@sholladay Do you?

Reply:
If `0` is fine, we can just remove the argument altogether.

Reply:
@sindresorhus https://github.com/sindresorhus/ky/pull/118

Reply:
I'm not sure if Chrome replaces `0` with `1` or no. In that case it wouldn't matter.